commercetools-models/src/main/java/io/sphere/sdk/carts/CartLike.java

```diff
@@ -200,4 +203,88 @@ default CurrencyUnit getCurrency() {
 
     @Nullable
     Locale getLocale();
+
+    /**
+     * Estimates the total price of the cart with taxes included, useful for B2C scenarios.
+     * This is the sum of the gross price of each item in the cart, shipping and discounts taken into account.
+     *
+     * Be aware that when taxes are not yet applied to the cart (i.e. shipping country/state are missing) then it cannot
+     * be determined whether taxes are included or not in the calculation, as it depends on the tax categories
+     * linked to the items in cart and the country/state of the cart.
+     *
+     * @return the estimated total gross price of the cart
+     */
+    default MonetaryAmount estimateTotalGrossPrice() {
+        return Optional.ofNullable(getTaxedPrice())
+                .map(TaxedPrice::getTotalGross)
+                .orElseGet(this::getTotalPrice);
+    }
```

Comment:
I feel that throwing an exception or `Oprtional.absent` is a better choice if we don't know yet whether taxes are included or not. I find it quite confusing if `estimateTotalGrossPrice` will return a net price in some circumstances.